models/intermediate/int_klaviyo__event_attribution.sql

```diff
@@ -0,0 +1,120 @@
+{{
+    config(
+        materialized='incremental',
+        unique_key='event_id',
+        partition_by={
+            "field": "occurred_on",
+            "data_type": "date"
+        } if target.type != 'spark' else ['occurred_on']
+    )
+}}
+
+with events as (
+
+    select 
+        *,
+        -- no event will be attributed to both a campaign and flow
+        coalesce(campaign_id, flow_id) as touch_id,
+        case 
+            when campaign_id is not null then 'campaign' 
+            when flow_id is not null then 'flow' 
+        else null end as touch_type -- defintion: touch = interaction with campaign/flow
+
+    from {{ var('event_table') }}
+
+    {% if is_incremental() %}
+    -- grab **ALL** events for users who have any events in this new increment
+    where person_id in (
+
+        select distinct person_id
+        from {{ var('event_table') }}
+
+        -- most events (from all kinds of integrations) at least once every hour
+        -- https://help.klaviyo.com/hc/en-us/articles/115005253208
+        where _fivetran_synced >= cast(coalesce( 
+            (
+                select {{ dbt_utils.dateadd(datepart = 'hour', 
+                                            interval = -1,
+                                            from_date_or_timestamp = 'max(_fivetran_synced)' ) }}  
+                from {{ this }}
+            ), '2010-01-01') as {{ dbt_utils.type_timestamp() }} )
```

Comment:
i just chose 2010-01-01 because i saw that date in a previous package 🤷 i guess klaviyo was only founded in 2012 so maybe i should change it to 2012-01-01??

Reply:
changed to 2012 🙂